dev/tools/localization/bin/gen_missing_localizations.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This program updates the language locale arb files with any missing resource
+// entries that are included in the english arb files. This is useful when
+// adding new resources for localization. You can just add the appropriate
+// entries to the english arb file and then run this script. It will then check
+// all of the other language locale arb files and update them with new 'TBD'
+// entries for any missing resources. These will be picked up by the localization
+// team and then translated.
+//
+// ## Usage
+//
+// Run this program from the root of the git repository.
+//
+// ```
+// dart dev/tools/localization/bin/gen_missing_localizations.dart
+// ```
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import '../localizations_utils.dart';
+import '../localizations_validator.dart';
+
+Future<void> main(List<String> rawArgs) async {
+  checkCwdIsRepoRoot('gen_missing_localizations');
+
+  final String l10nPath = path.join('packages', 'flutter_localizations', 'lib', 'src', 'l10n');
+  updateMissingResources(l10nPath, 'material');
+  updateMissingResources(l10nPath, 'cupertino');
+}
+
+Map<String, dynamic> loadBundle(File file) {
+  if (!FileSystemEntity.isFileSync(file.path))
+    exitWithError('Unable to find input file: ${file.path}');
+  return json.decode(file.readAsStringSync()) as Map<String, dynamic>;
+}
+
+void writeBundle(File file, Map<String, dynamic> bundle) {
+  final StringBuffer contents = StringBuffer();
+  contents.writeln('{');
+  for (final String key in bundle.keys) {
+    contents.writeln('  "$key": ${json.encode(bundle[key])}${key == bundle.keys.last ? '' : ','}');
+  }
+  contents.writeln('}');
+  file.writeAsStringSync(contents.toString());
+}
+
+Set<String> resourceKeys(Map<String, dynamic> bundle) {
+  return Set<String>.from(
+    // Skip any attribute keys
+    bundle.keys.where((String key) => !key.startsWith('@'))
+  );
+}
+
+bool intentionallyOmitted(String key, Map<String, dynamic> bundle) {
+  final String attributeKey = '@$key';
+  final dynamic attribute = bundle[attributeKey];
+  return attribute is Map && attribute.containsKey('notUsed');
+}
+
+/// Whether `key` corresponds to one of the plural variations of a key with
+/// the same prefix and suffix "Other".
+bool isPluralVariation(String key, Map<String, dynamic> bundle) {
+  final Match pluralMatch = kPluralRegexp.firstMatch(key);
+  if (pluralMatch == null)
+    return false;
+  final String prefix = pluralMatch[1];
+  return bundle.containsKey('${prefix}Other');
+}
+
+void updateMissingResources(String l10nPath, String groupPrefix) {
+  final Directory l10nDir = Directory(l10nPath);
```

Comment:
This would be easier to read if it were called "localizationDir". Same with "localizationPath"